test/Node/PackageJsonSpec.hs

```diff
@@ -50,11 +80,164 @@ packageTwo =
     , dependencyTags = Map.empty
     }
 
-spec :: Spec
-spec = do
+graphSpec :: Spec
+graphSpec =
   describe "buildGraph" $ do
     it "should produce expected output" $ do
       let graph = buildGraph mockInput
       expectDeps [packageOne, packageTwo] graph
       expectDirect [packageOne, packageTwo] graph
       expectEdges [] graph
+
+-- License Testing
+
+mockUrl1 :: Text
+mockUrl1 = "https://foo.com"
+
+mockUrl2 :: Text
+mockUrl2 = "https://bar.com"
+
+mockLicensesObjs :: [PkgJsonLicenseObj]
+mockLicensesObjs =
+  [ PkgJsonLicenseObj{licenseType = "MIT", licenseUrl = mockUrl1}
+  , PkgJsonLicenseObj{licenseType = "GPL", licenseUrl = mockUrl2}
+  ]
+
+mkTestLicenseResult :: FilePath -> [License] -> Types.LicenseResult
+mkTestLicenseResult manifest ls =
+  LicenseResult{licenseFile = manifest, licensesFound = ls}
+
+singleLicenseResult :: FilePath -> LicenseResult
+singleLicenseResult mockManifestFilePath =
+  mkTestLicenseResult mockManifestFilePath [License Types.UnknownType "MIT"]
+
+singleLicenseObjResult :: FilePath -> Text -> LicenseResult
+singleLicenseObjResult mockManifestFilePath url =
+  mkTestLicenseResult mockManifestFilePath [License LicenseURL url]
+
+multiLicenseObjResult :: FilePath -> [Text] -> LicenseResult
+multiLicenseObjResult mockManifestFilePath urls =
+  mkTestLicenseResult mockManifestFilePath $ License LicenseURL <$> urls
+
+mkMockPkgJsonGraph :: [(Manifest, PackageJson)] -> PkgJsonGraph
+mkMockPkgJsonGraph packagePairs =
+  PkgJsonGraph
+    { -- jsonGraph isn't relevant to license detection
+      jsonGraph = AM.empty
+    , jsonLookup = Map.fromList packagePairs
+    }
+
+mkNodeProject :: (PkgJsonGraph -> NodeProject) -> [(Manifest, PackageJson)] -> NodeProject
+mkNodeProject con = con . mkMockPkgJsonGraph
+
+licenseSpec :: Spec
+licenseSpec = do
+  -- The actual paths here don't matter much, but this is a cross-platform way
+  -- to get absolute filepaths.
+  currDir <- runIO getCurrentDir
+  let mockManifest = Manifest $ currDir </> $(mkRelFile "foo/package.json")
+      mockManifestFilePath = toFilePath . unManifest $ mockManifest
+
+  for_ [("NPM", NPM), ("Yarn", Yarn mockManifest), ("NPMLock", NPMLock mockManifest)] $ \(name, nodeConstr) ->
```

Comment:
[optional] [minor]

For this block, here are few low-hanging suggestions for readability: 

```
for_
    [ ("NPM", NPM)
    , ("Yarn", Yarn manifest)
    , ("NPMLock", NPMLock manifest)
    ]
    $ (...)
```

Since all of these are scoped in spec and in this module only, removing mock prefix would also help. 

- `mockUrl1 -> fooUrl`
- `mockManifest -> manifest`
- `mockManifestFilePath -> manifestPath`
- `foundLicenses -> licenses`
- `mockPackageJson -> packageJson`
 
Adding line break into some of long running line would also help

```haskell

      it' "It discovers an object license field" $ do
        let mockLicense = LicenseObj PkgJsonLicenseObj{licenseType = "MIT", licenseUrl = mockUrl1}
            mockPackageJson = pkgJsonMock (Just mockLicense) Nothing
        foundLicenses <- licenseAnalyzeProject . mkNodeProject nodeConstr $ [(mockManifest, mockPackageJson)]
        foundLicenses `shouldMatchList'` [singleLicenseObjResult mockManifestFilePath mockUrl1]

-- vs

        it' "should discovers an object license field" $ do
          let license = LicenseObj $ PkgJsonLicenseObj "MIT" fooUrl
              packageJson = mkPackageJson (Just license) Nothing
              expectedLicenses = [singleLicenseObjResult manifestPath fooUrl]

          licenses <-
            licenseAnalyzeProject
              . mkNodeProject nodeConstr
              $ [(manifest, packageJson)]

          licenses `shouldMatchList'` expectedLicenses
```